### ISSUE TITLE
Move the C-bindings to a separate repo

### ISSUE THREAD

**Issue by elBoberido** [eclipse-iceoryx/iceoryx#2030]
## Brief feature description

Similar to the DDS bridge (#1564) the C bindings are not a core part of iceoryx and can be moved to a separate repository.

## Detailed information

Moving the language bindings to separate repositories has the benefit of speeding up the work (and CI) in the core repo. The separate repo also opens the possibility to have non-core developer as maintainer of the specific language binding.

Ideally this is done right after the v3 release to have a nice transition path for user who are currently relying on the C bindings but are using master.

**Comment by elfenpiff** [comment]
@elBoberido makes sense. But I would like to have some kind of release process for this.

Meaning, what happens when iceoryx goes to the next version but all the language bindings/extensions are still hanging on the previous version? Should a release always include all the language bindings and they all have the same version - otherwise it would be maybe confusing that iceoryx-rs 2.0 works with iceoryx 3.5 or so.

So:
* versioning strategy for all bindings/extensions
* release strategy
* how to test that all bindings/extensions are working together?

**Comment by elBoberido** [comment]
@elfenpiff I thinks it would be okay if the version of the language binding does not follow the main project as long as the language bindings do not support all the features of the main project, i.e. the Rust bindings would remain at 0.something until they support all of the iceoryx features. After they achieve feature parity it would make sense to have them at least follow the major and minor version number. Having them to match the patch-level might be problematic when language binding specific bugfix releases are necessary.

For the release strategy I would suggest a best effort approach. After tagging the release candidate of the main project the maintainer of the respective language bindings should have 2-4 weeks time to update the bindings. If they are not updated it means there is no/little interest in the bindings and it should not be a blocker for the main project.

Testing could be done by a separate integration repo which would use launch testing or the robot framework to run multiple combinations of a hello world example.
